tests/test_main.py

```diff
@@ -103,3 +104,17 @@ def mocked_main(*args):
 
     with pytest.raises(FakeFailureError):
         xonsh.main.main()
+
+
+def test_xonsh_failback_script_from_file(shell, monkeypatch):
```

Comment:
"non-interactive should not fail back" is also a case, right?

Any suggests in code to write these cases without writting mocks?